### PR TITLE
Fix for issue #1287 - don't list hidden files in directory listing

### LINKED ISSUE #1287: Caddyfile in directory listing
(Are you asking for help with using Caddy? Please use our forum instead: https://forum.caddyserver.com. If you are filing a bug report, please answer the following questions. If your issue is not a bug report, you do not need to use this template. Either way, please consider donating if we've helped you. Thanks!)

### 1. What version of Caddy are you running (`caddy -version`)?
Caddy 0.9.3

### 2. What are you trying to do?

Show directory listing in root what have no index file.

### 3. What is your entire Caddyfile?
```text
localhost:8080

gzip
browse
ext    .html
log    access.log
```

### 4. How did you run Caddy (give the full command and describe the execution environment)?
```
theel0ja@mursu-3:~/Työpöytä/caddy$ caddy
```
Ubuntu 16.04
Installed via [getcaddy.com](https://getcaddy.com/)

### 5. What did you expect to see?
Caddyfile is hidden from directory listing, because it returns "404 Not Found"...
404 Not Found on /Caddyfile

### 6. What did you see instead (give full error messages and/or log)?
Directory listing with **Caddyfile**.
Directory listing on /

### 7. How can someone who is starting from scratch reproduce this behavior as minimally as possible?
Caddyfile to directory root with **browse** enabled in Caddyfile.

### PR DESCRIPTION
**Pull request by wendigo**
Hide hidden files in ```browse``` plugin, fixes #1287

### CODE REVIEW

**Comment by mholt** [comment]
Hey, this looks pretty good! I want to investigate that AppVeyor failure. Travis CI has a mechanism to overcome rate limits but I don't know how to do that with AppVeyor. I guess I could just trust that this works on Windows, but if there's a way to fix the CI environment here I think we should try to before merging -- or someone could confirm that it works on Windows for them.

**Comment by wendigo** [comment]
I think that the fix is to create `.gitcookies` file as described in: https://github.com/golang/go/issues/12933 (you know the details as I can see from discussion)

**Comment by mholt** [comment]
I know how to do it on Travis but not Appveyor... but it looks like it worked now. :)

**Comment by wendigo** [comment]
Yeah, it worked when I pulled master changes into branch :)

File: caddyhttp/browse/browse.go [hunk 1]
```diff
@@ -212,6 +219,10 @@ func directoryListing(files []os.FileInfo, canGoUp bool, urlPath string) (Listin
 
 		url := url.URL{Path: "./" + name} // prepend with "./" to fix paths with ':' in the name
 
+		if _, exists := hidden[f.Name()]; exists {
```
Comment (mholt) [inline caddyhttp/browse/browse.go]: Does this take into account different representations of the file paths?

Maybe we should export and use https://github.com/mholt/caddy/blob/8653b70c32f9a79f165d9417f4c6c45eebbc6dd1/caddyhttp/staticfiles/fileserver.go#L136 instead.

Comment (wendigo) [inline caddyhttp/browse/browse.go]: Nope. I haven't seen isHidden function - I will look into it! Thx for a tip :)

**Comment by wendigo** [comment]
@mholt: I've exported and used IsHidden

**Pull request merged**